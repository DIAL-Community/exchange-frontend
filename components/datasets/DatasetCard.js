import Link from 'next/link'
import { useIntl } from 'react-intl'
import { useCallback, useEffect } from 'react'
import ReactTooltip from 'react-tooltip'
import Image from 'next/image'
import { ORIGIN_ACRONYMS, ORIGIN_EXPANSIONS } from '../../lib/utilities'

const ellipsisTextStyle = `
  whitespace-nowrap text-ellipsis overflow-hidden my-auto
`
const containerElementStyle = `
  border-3 cursor-pointer
  border-transparent hover:border-dial-yellow
  text-dataset hover:text-dial-yellow
`

const DatasetCard = ({ dataset, listType, newTab = false }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  useEffect(() => {
    ReactTooltip.rebuild()
  })

  return (
    <Link href={`/datasets/${dataset.slug}`}>
      <a {... newTab && { target: '_blank' }}>
        {
          listType === 'list'
            ? (
              <div className={`${containerElementStyle}`}>
                <div className='bg-white border border-dial-gray hover:border-transparent card-drop-shadow'>
                  <div className='flex flex-row flex-wrap lg:gap-x-4 px-4' style={{ minHeight: '4.5rem' }}>
                    <div className='w-10/12 lg:w-4/12 text-base font-semibold text-dial-gray-dark my-auto relative'>
                      <Image
                        layout='fill'
                        objectFit='scale-down'
                        objectPosition='left'
                        sizes='1vw'
                        alt={format('image.alt.logoFor', { name: dataset.name })}
                        src={process.env.NEXT_PUBLIC_GRAPHQL_SERVER + dataset.imageFile}
                      />
                      <div className={`ml-8 w-4/5 h-3/5 font-semibold relative ${ellipsisTextStyle}`}>
                        {dataset.name}
                      </div>
                    </div>
                    <div className={`w-8/12 lg:w-4/12 text-sm lg:text-base text-dial-purple ${ellipsisTextStyle}`}>
                      {dataset.origins && dataset.origins.length === 0 && format('general.na')}
                      {
                        dataset.origins && dataset.origins.length > 0 &&
                        dataset.origins
                          .map(origin => ORIGIN_EXPANSIONS[origin.name.toLowerCase()] || origin.name)
                          .join(', ')
                      }
                    </div>
                    <div className='w-4/12 lg:w-2/12 text-right text-sm lg:text-base font-semibold text-dial-cyan my-auto'>
                      {dataset.datasetType === 'dataset' ? format('dataset.card.dataset').toUpperCase() : ''}
                    </div>
                  </div>
                </div>
              </div>
            )
            : (
              <div className={containerElementStyle}>
                <div className='h-full flex flex-col border border-dial-gray hover:border-transparent card-drop-shadow'>
                  <div className='flex flex-row gap-x-1.5 p-1.5 border-b border-dial-gray dataset-card-header'>
                    {
                      dataset.tags.indexOf(format('dataset.card.coronavirusTagValue').toLowerCase()) >= 0 &&
                        <img
                          alt={format('image.alt.logoFor', { name: format('coronavirus.title') })}
                          data-tip={format('tooltip.covid')} className='h-5' src='/icons/coronavirus/coronavirus.png'
                        />
                    }
                    {
                      <div className='ml-auto text-dial-cyan text-sm font-semibold'>
                        {dataset.datasetType.toUpperCase()}
                      </div>
                    }
                  </div>
                  <div className='flex flex-col h-80 p-4'>
                    <div className='text-2xl font-semibold absolute w-64 2xl:w-80'>
                      {dataset.name}
                    </div>
                    {
                      dataset.datasetDescription &&
                        <img
                          className='ml-auto opacity-20 hover:opacity-100 dataset-filter'
                          data-tip={dataset.datasetDescription.description}
                          data-html
                          alt='Info' height='20px' width='20px' src='/icons/info.svg'
                        />
                    }
                    <div className='m-auto w-3/5 h-3/5 relative' >
                      <Image
                        layout='fill'
                        objectFit='contain'
                        alt={format('image.alt.logoFor', { name: dataset.name })}
                        src={process.env.NEXT_PUBLIC_GRAPHQL_SERVER + dataset.imageFile}
                      />
                    </div>
                  </div>
                  <div className='flex flex-col bg-dial-gray-light text-dial-gray-dark mt-auto'>
                    <div className='pb-3 flex flex-row flex-wrap justify-between border-b border-dial-gray'>
                      <div className='pl-3 pt-3 flex flex-row'>
                        <div className='text-base my-auto mr-1'>{format('dataset.card.sdgs')}</div>
                        <div className='bg-white rounded p-1.5 flex flex-row gap-x-1'>
                          {
                            dataset.sustainableDevelopmentGoals.length === 0 &&
                              <span className='text-base my-1 mx-auto font-semibold'>
                                {format('general.na')}
                              </span>
                          }
                          {
                            dataset.sustainableDevelopmentGoals
                              .filter((_, index) => index <= 1)
                              .map(sdg => (
                                <img
                                  data-tip={format('tooltip.forEntity', { entity: format('sdg.label'), name: sdg.name })}
                                  key={`sdg-${sdg.slug}`} className='h-8 cursor-default'
                                  alt={format('image.alt.logoFor', { name: sdg.name })}
                                  src={`/images/sdgs/${sdg.slug}.png`}
                                />
                              ))
                          }
                          {
                            dataset.sustainableDevelopmentGoals.length > 2 &&
                              <span className='text-base'>...</span>
                          }
                        </div>
                      </div>
                    </div>
                    <div className='flex flex-row justify-between text-dial-gray-dark'>
                      <div className='pl-3 py-3 flex-auto flex flex-col'>
                        <div className='text-base my-auto'>{format('product.card.license')}</div>
                        <div className='bg-white mt-1.5 mr-auto p-2 rounded text-sm font-semibold'>
                          {(dataset.license || format('general.na')).toUpperCase()}
                        </div>
                      </div>
                      <div className='pr-3 py-3 flex-auto flex flex-col'>
                        <div className='text-base text-right my-auto'>Sources</div>
                        <div className='flex flex-row justify-end font-semibold'>
                          {
                            dataset.origins.length === 0 &&
                              <div className='bg-white mt-1.5 mr-1.5 last:mr-0 p-2 rounded text-sm'>
                                {format('general.na')}
                              </div>
                          }
                          {
                            dataset.origins
                              .filter((_, index) => index <= 2)
                              .map(origin => {
                                return (
                                  <div
                                    key={`origin-${origin.slug}`} className='bg-white mt-1.5 mr-1.5 last:mr-0 p-2 rounded text-sm'
                                  >
                                    {(ORIGIN_ACRONYMS[origin.slug.toLowerCase()] || origin.slug).toUpperCase()}
                                  </div>
                                )
                              })
                          }
                          {
                            dataset.origins.length > 3 &&
                              <div
                                className='bg-white mt-1.5 mr-1.5 last:mr-0 p-2 rounded text-sm'
                                data-tip={format('tooltip.ellipsisFor', { entity: format('dataset.label') })}
                              >
                                &hellip;
                              </div>
                          }
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            )
        }
      </a>
    </Link>
  )
}

export default DatasetCard
