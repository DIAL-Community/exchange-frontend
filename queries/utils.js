const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMature = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input,
    mature: true
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}
