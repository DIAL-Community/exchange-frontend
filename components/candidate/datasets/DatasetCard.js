import React, { useCallback, useMemo, useState } from 'react'
import { FaRegCheckCircle, FaRegTimesCircle } from 'react-icons/fa'
import { useIntl } from 'react-intl'
import { useMutation } from '@apollo/client'
import { useRouter } from 'next/router'
import parse from 'html-react-parser'
import classNames from 'classnames'
import { getDatasetTypeOptions, prependUrlWithProtocol } from '../../../lib/utilities'
import { CANDIDATE_DATASET_ACTION } from '../../../queries/candidate'
import { useUser } from '../../../lib/hooks'
import { CandidateActionType, CandidateStatusType } from '../../../lib/constants'

const hoverEffectTextStyle = 'border-b-2 border-transparent hover:border-dial-sunshine'

const DatasetCard = ({ dataset, listType }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const [loading, setLoading] = useState(false)
  const [status, setStatus] = useState('')
  const [comment, setComment] = useState('')

  const datasetType = useMemo(() =>
    getDatasetTypeOptions(format)
      .find((datasetOption) => dataset.dataType === datasetOption.value)
  , [dataset.dataType, format])

  const shouldFlipCard = (status) =>
    [CandidateStatusType.REJECTION, CandidateStatusType.APPROVAL].indexOf(status) >= 0

  return (
    <>
      {
        listType === 'list'
          ? (
            <div className={classNames('card', { 'flip-horizontal':  shouldFlipCard(status) })}>
              <div className='card-body border-3 border-transparent'>
                <div className={classNames('card-front border border-dial-gray shadow-md bg-white',
                  { 'bg-red-50': dataset.rejected === true || status === CandidateStatusType.REJECTED },
                  { 'bg-emerald-50': dataset.rejected === false || status === CandidateStatusType.APPROVED }
                )}>
                  <div className='flex flex-col xl:flex-row gap-3 p-3'>
                    <div className='flex flex-col gap-3 w-full xl:w-2/3 '>
                      <div className='my-auto line-clamp-1 font-semibold'>
                        {dataset.name}
                      </div>
                      <div className='my-auto line-clamp-1 text-dial-blue'>
                        <a href={prependUrlWithProtocol(dataset.dataUrl)} target='_blank' rel='noreferrer'>
                          {dataset.dataUrl} ⧉
                        </a>
                      </div>
                      <div className='line-clamp-2'>
                        {dataset.description && parse(dataset.description)}
                      </div>
                      <div className='line-clamp-1'>
                        {dataset.submitterEmail}
                      </div>
                    </div>
                    <div className='absolute right-3 text-sm text-dial-cyan font-semibold'>
                      <div className='px-2 py-1 border border-dial-cyan rounded'>
                        {datasetType.label.toUpperCase()}
                      </div>
                    </div>
                    <div className='ml-auto mt-auto'>
                      {
                        dataset.rejected === null &&
                        status !== CandidateStatusType.APPROVED &&
                        status !== CandidateStatusType.REJECTED &&
                          <div className='ml-auto flex flex-row gap-3'>
                            <ToggleRejectionButton {...{ setStatus, loading }} style='text-sm secondary-button' />
                            <ToggleApprovalButton {...{ setStatus, loading }} style='text-sm submit-button' />
                          </div>
                      }
                    </div>
                  </div>
                </div>
                <div className={classNames(
                  'card-back flip-horizontal border border-dial-gray shadow-sm bg-dial-gray h-full',
                  { 'bg-red-50': dataset.rejected === true || status === CandidateStatusType.REJECTED },
                  { 'bg-emerald-50': dataset.rejected === false || status === CandidateStatusType.APPROVED }
                )}>
                  <div className='flex flex-col gap-3 p-3 h-full'>
                    <div className='flex flex-row xl:hidden text-dial-gray-dark text-sm font-semibold'>
                      <div className='ml-auto my-auto p-1.5 border border-dial-gray-dark rounded'>
                        {datasetType.label.toUpperCase()}
                      </div>
                    </div>
                    <label className='block'>
                      <span className='sr-only text-gray-700'>{format('candidate.feedback')}</span>
                      <input
                        className='w-full'
                        type='text'
                        placeholder={format('candidate.feedback.placeholder')}
                        value={comment}
                        onChange={(e) => setComment(e.target.value)}
                      />
                    </label>
                    <div className='ml-auto mt-auto flex flex-row gap-3'>
                      {
                        status === CandidateStatusType.REJECTION &&
                        <>
                          <CancelButton {...{ status, setStatus, loading }} />
                          <DeclineButton {...{ status, setStatus, loading, setLoading, dataset }} />
                        </>
                      }
                      {
                        status === CandidateStatusType.APPROVAL &&
                        <>
                          <CancelButton {...{ status, setStatus, loading }} />
                          <ApproveButton {...{ status, setStatus, loading, setLoading, dataset }} />
                        </>
                      }
                    </div>
                  </div>
                </div>
              </div>
            </div>
          )
          : (
            <div className={classNames('card', { 'flip-vertical':  shouldFlipCard(status) })}>
              <div className='card-body border-3 border-transparent text-dial-purple h-full'>
                <div className='card-front h-full flex flex-col border border-dial-gray shadow-md'>
                  <div className='flex flex-row p-1.5 border-b border-dial-gray dataset-card-header'>
                    {
                      (dataset.rejected === true || status === CandidateStatusType.REJECTED) &&
                        <div className='bg-red-500 py-1 px-2 rounded text-white text-sm font-semibold'>
                          {format('candidate.rejected').toUpperCase()}
                        </div>
                    }
                    {
                      (dataset.rejected === false || status === CandidateStatusType.APPROVED) &&
                        <div className='bg-emerald-500 py-1 px-2 rounded text-white text-sm font-semibold'>
                          {format('candidate.approved').toUpperCase()}
                        </div>
                    }
                    {
                      dataset.rejected === null &&
                        <div className='ml-auto my-auto py-1 px-2 text-dial-cyan text-sm font-semibold'>
                          {format('candidateDataset.label').toUpperCase()}
                        </div>
                    }
                  </div>
                  <div className='flex flex-col h-64 p-4'>
                    <div className='text-2xl font-semibold'>
                      {dataset.name}
                    </div>
                    <div className='mt-3 line-clamp-6'>
                      {dataset.description && parse(dataset.description)}
                    </div>
                  </div>
                  <div className='py-3 flex flex-col gap-3 bg-dial-gray-light text-dial-gray-dark mt-auto'>
                    <div className='px-3 ml-auto'>
                      {dataset.submitterEmail}
                    </div>
                    <div className='px-3 ml-auto flex flex-row'>
                      {
                        dataset.dataUrl
                          ? (
                            <a
                              href={prependUrlWithProtocol(dataset.dataUrl)}
                              target='_blank'
                              rel='noreferrer'
                              className={'py-1 px-2 bg-white text-dial-blue break-all line-clamp-1'}
                            >
                              <span className={hoverEffectTextStyle}>
                                {dataset.dataUrl} ⧉
                              </span>
                            </a>
                          )
                          : (
                            <div className='py-1 px-2 bg-white line-clamp-1'>
                              {format('general.na')}
                            </div>
                          )
                      }
                    </div>
                    <div className='px-3 my-auto flex flex-row'>
                      <div className='ml-auto px-2 py-1 bg-white'>
                        {datasetType.label}
                      </div>
                    </div>
                    {
                      dataset.rejected === null &&
                      status !== CandidateStatusType.APPROVED &&
                      status !== CandidateStatusType.REJECTED &&
                        <>
                          <div className='border-t'></div>
                          <div className='ml-auto flex flex-row gap-3 mx-3'>
                            <ToggleRejectionButton {...{ setStatus, loading }} style='text-sm secondary-button' />
                            <ToggleApprovalButton {...{ setStatus, loading }} style='text-sm submit-button' />
                          </div>
                        </>
                    }
                  </div>
                </div>
                <div className='card-back flip-vertical flex flex-col border border-dial-gray bg-dial-gray shadow-lg h-full'>
                  <div className='flex flex-row p-1.5 bg-dial-gray-dark'>
                    <div className='ml-auto my-auto text-dial-cyan text-sm font-semibold'>
                      {format('candidateDataset.label').toUpperCase()}
                    </div>
                  </div>
                  <div className='flex flex-col gap-3 p-3 h-full'>
                    <label className='block'>
                      <span className='text-gray-700'>{format('candidate.feedback')}</span>
                      <textarea
                        className='w-full'
                        rows={8}
                        style={{ resize: 'none' }}
                        placeholder={format('candidate.feedback.placeholder')}
                        value={comment}
                        onChange={(e) => setComment(e.target.value)}
                      />
                    </label>
                    <div className='ml-auto mt-auto flex flex-row gap-3'>
                      {
                        status === CandidateStatusType.REJECTION &&
                          <>
                            <CancelButton {...{ status, setStatus, loading }} />
                            <DeclineButton {...{ status, setStatus, loading, setLoading, dataset }} />
                          </>
                      }
                      {
                        status === CandidateStatusType.APPROVAL &&
                        <>
                          <CancelButton {...{ status, setStatus, loading }} />
                          <ApproveButton {...{ status, setStatus, loading, setLoading, dataset }} />
                        </>
                      }
                    </div>
                  </div>
                </div>
              </div>
            </div>
          )
      }
    </>
  )
}

const CancelButton = ({ status, setStatus, loading }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  return (
    <button
      className={classNames(
        'text-sm secondary-button',
        { 'block': status !== '' }, { 'hidden': status === '' }
      )}
      onClick={() => setStatus('')}
      disabled={loading}
    >
      {format('candidate.cancel')}
    </button>
  )
}

const ToggleApprovalButton = ({ style, setStatus, loading }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  return (
    <button
      className={style}
      onClick={() => setStatus(CandidateStatusType.APPROVAL)}
      disabled={loading}
    >
      {format('candidate.approve')}
      <FaRegCheckCircle className='ml-1 inline text-xl' />
    </button>
  )
}

const ApproveButton = ({ dataset, setStatus, loading, setLoading }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { locale } = useRouter()
  const { user } = useUser()

  const [candidateDatasetApproval, { reset }] = useMutation(CANDIDATE_DATASET_ACTION, {
    onCompleted: (data) => {
      const { approveRejectCandidateDataset: response } = data
      if (response?.candidateDataset && response?.errors?.length === 0) {
        setStatus(CandidateStatusType.APPROVED)
        setLoading(false)
      } else {
        setLoading(false)
        reset()
      }
    },
    onError: () => {
      setLoading(false)
      reset()
    }
  })

  const approveDataset = async () => {
    if (user) {
      setLoading(true)
      const { userEmail, userToken } = user

      candidateDatasetApproval({
        variables: {
          slug: dataset.slug,
          action: CandidateActionType.APPROVE
        },
        context: {
          headers: {
            'Accept-Language': locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  return (
    <button className='text-sm submit-button' onClick={approveDataset} disabled={loading}>
      {format('candidate.approve')}<FaRegCheckCircle className='ml-1 inline text-xl' />
    </button>
  )
}

const ToggleRejectionButton = ({ style, setStatus, loading }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  return (
    <button
      className={style}
      onClick={() => setStatus(CandidateStatusType.REJECTION)}
      disabled={loading}
    >
      {format('candidate.reject')}
      <FaRegTimesCircle className='ml-1 inline text-xl text-red-500' />
    </button>
  )
}

const DeclineButton = ({ dataset, setStatus, loading, setLoading }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { locale } = useRouter()
  const { user } = useUser()

  const [candidateDatasetApproval, { reset }] = useMutation(CANDIDATE_DATASET_ACTION, {
    onCompleted: (data) => {
      const { approveRejectCandidateDataset: response } = data
      if (response?.candidateDataset && response?.errors?.length === 0) {
        setStatus(CandidateStatusType.REJECTED)
        setLoading(false)
      } else {
        setLoading(false)
        reset()
      }
    },
    onError: () => {
      setLoading(false)
      reset()
    }
  })

  const rejectDataset = async () => {
    if (user) {
      setLoading(true)
      const { userEmail, userToken } = user
      const variables = {
        slug: dataset.slug,
        action: CandidateActionType.REJECT
      }

      candidateDatasetApproval({
        variables,
        context: {
          headers: {
            'Accept-Language': locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  return (
    <button className='text-sm submit-button' onClick={rejectDataset} disabled={loading}>
      {format('candidate.reject')}
      <FaRegTimesCircle className='ml-1 inline text-xl text-red-500' />
    </button>
  )
}

export default DatasetCard
