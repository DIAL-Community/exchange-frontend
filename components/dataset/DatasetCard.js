import { useCallback } from 'react'
import classNames from 'classnames'
import parse from 'html-react-parser'
import Link from 'next/link'
import { FaXmark } from 'react-icons/fa6'
import { useIntl } from 'react-intl'
import { DisplayType } from '../utils/constants'
import { isValidFn } from '../utils/utilities'

const DatasetCard = ({ displayType, index, dataset, dismissHandler }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const displayLargeCard = () =>
    <div className={`px-4 py-6 rounded-lg min-h-[13.5rem] ${index % 2 === 0 && 'bg-dial-violet'}`}>
      <div className='flex flex-col lg:flex-row gap-x-6 gap-y-3'>
        {dataset.imageFile.indexOf('placeholder.svg') < 0 &&
          <div className='w-16 h-16 mx-auto bg-white border'>
            <img
              src={process.env.NEXT_PUBLIC_GRAPHQL_SERVER + dataset.imageFile}
              alt={format('ui.image.logoAlt', { name: format('ui.dataset.label') })}
              className='object-contain w-10 h-10 mx-auto my-2'
            />
          </div>
        }
        {dataset.imageFile.indexOf('placeholder.svg') >= 0 &&
          <div className='w-16 h-16 mx-auto bg-dial-plum rounded-full'>
            <img
              src={process.env.NEXT_PUBLIC_GRAPHQL_SERVER + dataset.imageFile}
              alt={format('ui.image.logoAlt', { name: format('ui.dataset.label') })}
              className='object-contain w-10 h-10 white-filter mt-3 mx-auto'
            />
          </div>
        }
        <div className='flex flex-col gap-y-3 max-w-3xl lg:w-10/12'>
          <div className='text-lg font-semibold text-dial-plum'>
            {dataset.name}
          </div>
          <div className='line-clamp-4 text-dial-stratos'>
            {dataset?.datasetDescription && parse(dataset?.parsedDescription)}
          </div>
          <div className='flex gap-x-2 text-dial-stratos'>
            <div className='text-sm'>
              {format('ui.sdg.header')} ({dataset.sdgs?.length ?? 0})
            </div>
            <div className='border-r border-dial-slate-400' />
            <div className='text-sm'>
              {format('ui.sector.header')} ({dataset.sectors?.length ?? 0})
            </div>
            <div className='border-r border-dial-slate-400' />
            <div className='text-sm'>
              {format('ui.tag.header')} ({dataset.tags?.length ?? 0})
            </div>
          </div>
        </div>
      </div>
    </div>

  const displaySmallCard = () =>
    <div className='rounded-lg bg-gradient-to-r from-dataset-bg-light to-dataset-bg h-16'>
      <div className='flex flex-row gap-x-3 px-6 h-full'>
        {dataset.imageFile.indexOf('placeholder.svg') >= 0 &&
          <div className='rounded-full my-auto w-10 h-10 min-w-[2.5rem]'>
            <img
              src={process.env.NEXT_PUBLIC_GRAPHQL_SERVER + dataset.imageFile}
              alt={format('ui.image.logoAlt', { name: format('ui.dataset.header') })}
              className='object-contain w-10 h-10 my-auto'
            />
          </div>
        }
        {dataset.imageFile.indexOf('placeholder.svg') < 0 &&
          <img
            src={process.env.NEXT_PUBLIC_GRAPHQL_SERVER + dataset.imageFile}
            alt={format('ui.image.logoAlt', { name: format('ui.dataset.header') })}
            className='object-contain w-10 h-10 my-auto min-w-[2.5rem]'
          />
        }
        <div className='text-sm font-semibold text-dial-plum my-auto'>
          {dataset.name}
        </div>
      </div>
    </div>

  const displayGridCard = () => (
    <div className='cursor-pointer hover:rounded-lg hover:shadow-lg'>
      <div className='bg-white shadow-lg rounded-xl h-[360px]'>
        <div className="flex flex-col h-full">
          <div
            className={
              classNames(
                'flex justify-center items-center bg-white',
                'rounded-xl border-4 border-dial-warm-beech',
                'py-12 mx-4 my-4 max-h-[180px]'
              )}
          >
            {dataset.imageFile.indexOf('placeholder.svg') < 0 &&
              <div className="inline my-4 mx-6">
                <img
                  src={process.env.NEXT_PUBLIC_GRAPHQL_SERVER + dataset.imageFile}
                  alt={format('ui.image.logoAlt', { name: format('ui.product.label') })}
                  className="object-contain max-h-[80px]" width="80px"
                />
              </div>
            }
            {dataset.imageFile.indexOf('placeholder.svg') >= 0 &&
              <div className="w-20 h-20 flex justify-center items-center">
                <img
                  src={process.env.NEXT_PUBLIC_GRAPHQL_SERVER + dataset.imageFile}
                  alt={format('ui.image.logoAlt', { name: format('ui.product.label') })}
                  className='object-contain w-10 h-10 mx-auto'
                />
              </div>
            }
          </div>
          <div className="px-6 text-xl text-center font-semibold line-clamp-1">
            {dataset.name}
          </div>
          <div className="px-6 py-2 text-xs text-dial-stratos font-medium">
            <span className="text-center line-clamp-3">
              {dataset?.datasetDescription && parse(dataset?.parsedDescription)}
            </span>
          </div>
          <div className="my-3 mx-auto text-xs font-medium">
            <div className="rounded-full bg-dial-orange uppercase shadow-none px-6 py-1 text-white">
              {dataset?.datasetType}
            </div>
          </div>
        </div>
      </div>
    </div>
  )

  return (
    <div className='relative'>
      <Link href={`/datasets/${dataset.slug}`}>
        {displayType === DisplayType.GRID_CARD && displayGridCard()}
        {displayType === DisplayType.LARGE_CARD && displayLargeCard()}
        {displayType === DisplayType.SMALL_CARD && displaySmallCard()}
      </Link>
      { isValidFn(dismissHandler) &&
        <button type='button' className='absolute top-2 right-2'>
          <FaXmark size='1rem' className='text-dial-plum' onClick={dismissHandler} />
        </button>
      }
    </div>
  )
}

export default DatasetCard
