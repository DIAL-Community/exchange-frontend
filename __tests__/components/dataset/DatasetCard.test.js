import { render } from '../../test-utils'
import { mockNextUseRouter } from '../../utils/nextMockImplementation'
import DatasetCard from '../../../components/datasets/DatasetCard'
import { dataset } from './data/DatasetCard'

mockNextUseRouter()

describe('Unit tests for the DatasetCard component', () => {
  describe('Should match snapshot -', () => {
    test('list mode', () => {
      const { container } = render(<DatasetCard dataset={dataset} listType='list' />)
      expect(container).toMatchSnapshot()
    })

    test('card mode', () => {
      const { container } = render(<DatasetCard dataset={dataset} />)
      expect(container).toMatchSnapshot()
    })
  })
})
