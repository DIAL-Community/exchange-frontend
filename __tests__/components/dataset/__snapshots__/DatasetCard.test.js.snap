// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the DatasetCard component Should match snapshot - card mode 1`] = `
<div>
  <div
    class="Toastify"
  />
  <a
    href="/datasets/test_dataset"
  >
    <div
      class="
  border-3 cursor-pointer
  border-transparent hover:border-dial-yellow
  text-dataset hover:text-dial-yellow
"
    >
      <div
        class="h-full flex flex-col border border-dial-gray hover:border-transparent card-drop-shadow"
      >
        <div
          class="flex flex-row gap-x-1.5 p-1.5 border-b border-dial-gray dataset-card-header"
        >
          <img
            alt="Logo for: COVID-19"
            class="h-5"
            data-tip="This product has been tagged as relevant for COVID response,"
            src="/icons/coronavirus/coronavirus.png"
          />
          <div
            class="ml-auto text-dial-cyan text-sm font-semibold"
          >
            DATASET
          </div>
        </div>
        <div
          class="flex flex-col h-80 p-4"
        >
          <div
            class="text-2xl font-semibold absolute w-64 2xl:w-80"
          >
            Test Dataset
          </div>
          <img
            alt="Info"
            class="ml-auto opacity-20 hover:opacity-100 dataset-filter"
            data-html="true"
            data-tip="Test Description"
            height="20px"
            src="/icons/info.svg"
            width="20px"
          />
          <div
            class="m-auto w-3/5 h-3/5 relative"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Logo for: Test Dataset"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
              />
              <noscript />
            </span>
          </div>
        </div>
        <div
          class="flex flex-col bg-dial-gray-light text-dial-gray-dark mt-auto"
        >
          <div
            class="pb-3 flex flex-row flex-wrap justify-between border-b border-dial-gray"
          >
            <div
              class="pl-3 pt-3 flex flex-row"
            >
              <div
                class="text-base my-auto mr-1"
              >
                SDGs
              </div>
              <div
                class="bg-white rounded p-1.5 flex flex-row gap-x-1"
              >
                <img
                  alt="Logo for: Test SDG"
                  class="h-8 cursor-default"
                  data-tip="Sustainable Development Goal: Test SDG"
                  src="/images/sdgs/test_sdg.png"
                />
              </div>
            </div>
          </div>
          <div
            class="flex flex-row justify-between text-dial-gray-dark"
          >
            <div
              class="pl-3 py-3 flex-auto flex flex-col"
            >
              <div
                class="text-base my-auto"
              >
                License
              </div>
              <div
                class="bg-white mt-1.5 mr-auto p-2 rounded text-sm font-semibold"
              >
                TEST LICENCE
              </div>
            </div>
            <div
              class="pr-3 py-3 flex-auto flex flex-col"
            >
              <div
                class="text-base text-right my-auto"
              >
                Sources
              </div>
              <div
                class="flex flex-row justify-end font-semibold"
              >
                <div
                  class="bg-white mt-1.5 mr-1.5 last:mr-0 p-2 rounded text-sm"
                >
                  TESTORIGIN
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;

exports[`Unit tests for the DatasetCard component Should match snapshot - list mode 1`] = `
<div>
  <div
    class="Toastify"
  />
  <a
    href="/datasets/test_dataset"
  >
    <div
      class="
  border-3 cursor-pointer
  border-transparent hover:border-dial-yellow
  text-dataset hover:text-dial-yellow
"
    >
      <div
        class="bg-white border border-dial-gray hover:border-transparent card-drop-shadow"
      >
        <div
          class="flex flex-row flex-wrap lg:gap-x-4 px-4"
          style="min-height: 4.5rem;"
        >
          <div
            class="w-10/12 lg:w-4/12 text-base font-semibold text-dial-gray-dark my-auto relative"
          >
            <span
              style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
            >
              <img
                alt="Logo for: Test Dataset"
                data-nimg="fill"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: scale-down; object-position: left;"
              />
              <noscript />
            </span>
            <div
              class="ml-8 w-4/5 h-3/5 font-semibold relative 
  whitespace-nowrap text-ellipsis overflow-hidden my-auto
"
            >
              Test Dataset
            </div>
          </div>
          <div
            class="w-8/12 lg:w-4/12 text-sm lg:text-base text-dial-purple 
  whitespace-nowrap text-ellipsis overflow-hidden my-auto
"
          >
            Test origin
          </div>
          <div
            class="w-4/12 lg:w-2/12 text-right text-sm lg:text-base font-semibold text-dial-cyan my-auto"
          >
            DATASET
          </div>
        </div>
      </div>
    </div>
  </a>
</div>
`;
